### DISCUSSION TITLE
Elasticsearch Upgrade Guide Needed for 6.x to 7.x

### DISCUSSION THREAD

**Discussion by ztaylor100** [mastodon/mastodon#17916] [Q&A]
In consideration that the official Mastodon documentation includes the installation of Elasticsearch 6.x, is anyone able to provide an easy-to-follow guide on how to upgrade to a supported version for Mastodon 3.5.0? For a single node instance, I would think It HAS to be easier than the rolling upgrade instructions in the official ES documentation. The internet is not being very helpful! Thanks in advance.

**Comment by marians** [comment]
Still in need?

We are running on docker-compose.

First I stopped ES.

Basically I adopted the service config given in https://github.com/mastodon/mastodon/blob/main/docker-compose.yml#L26-L55 

I had to make two changes:

- remove the three lines at/under `nofile`, as these caused errors.
- changed the image tag from `7.17.4` to `7.17.5`

Next I had to execute

```
sysctl -w vm.max_map_count=262144
```

on the host, to mitigate this error:

```
bootstrap check failure [1] of [1]: max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
```

After that I wiped the old Elasticsearch data dir. That should NOT have been necessary, but I preferred a clean start.

Then started ES and waited for it to be reported as healthy.

Finally I ran `tootctl search deploy` to create the index. This took several hours for ~ 3 million records.
